spec/features/users/sign_up_spec.rb

```diff
@@ -23,6 +23,21 @@
     end
   end
 
+  context 'user cancels on the enter password screen' do
+    it 'deletes the user record and returns them to the home page' do
+      email = 'test@test.com'
+
+      visit sign_up_email_path
+
+      submit_form_with_valid_email
+      click_confirmation_link_in_email(email)
+      click_on t('links.cancel_account_creation')
+
+      expect(current_path).to eq root_path
+      expect(User.find_with_email(email)).to be_nil
```

Comment:
The framework I follow for testing is:

- Feature spec: end to end tests with as little stubbing as possible. Actions taken are via browser (clicking around like a user would). Expectations are from the perspective of the user (user would not know about database changes).

- Controller tests for testing multiple paths through a controller (faster than feature specs, so usually use these to test 'failure' cases as opposed to creating a feature for every possible scenario)

- Unit tests for methods

This is in line with https://robots.thoughtbot.com/how-we-test-rails-applications